### PR TITLE
Cardano cluster, on-fly genesis, Chairman service & NixOS test

### PR DESCRIPTION
**Pull request by deepfire**
1. Turn the `cardano-node.service` into instanced set of `cardano-node@NODE-ID.service`, systemd-unit-wise
2. Define a `cardano-cluster` target.
3. Generate genesis on fly for the cluster.
4. Chairman service, with a factored script generator..
5. Chairman's cluster NixOS test.

### CODE REVIEW

File: nix/nixos/cardano-node-options.nix [hunk 1]
```diff
@@ -109,7 +110,7 @@ in {
 
       stateDir = mkOption {
         type = types.str;
-        default = "/var/lib/cardano-node";
+        default = "/var/lib/cardano-node-%i";
```
Comment (disassembler) [inline nix/nixos/cardano-node-options.nix]: lets not pollute lib. Maybe optionally if multiple instances are running, do `/var/lib/cardano-node/%i`

File: nix/nixos/cardano-node-service.nix [hunk 2]
```diff
@@ -27,7 +27,7 @@ in {
       uid = 10016;
       group = "cardano-node";
     };
-    systemd.services.cardano-node = {
+    systemd.services."cardano-node@" = {
```
Comment (disassembler) [inline nix/nixos/cardano-node-service.nix]: we should optionally do this if it's a cluster vs just a single node

**Comment by deepfire** [comment]
> I don't see how this can be ran without having systemd

Absolutely, it needs systemd.

File: nix/nixos/cardano-node-options.nix (hunk 1 above)
Comment (deepfire) [reply nix/nixos/cardano-node-options.nix]: Will do!

File: nix/nixos/cardano-node-service.nix (hunk 2 above)
Comment (deepfire) [reply nix/nixos/cardano-node-service.nix]: Agreed.

File: nix/nixos/cardano-node-options.nix (hunk 1 above)
Comment (deepfire) [reply nix/nixos/cardano-node-options.nix]: Fixed.

**Comment by disassembler** [comment]
Breaks connect scripts:
```
nix-build -A scripts.mainnet.node
error: The option `services.cardano-node.port' defined in `<unknown-file>' does not exist.
```

**Comment by deepfire** [comment]
> Breaks the scripts.

Hmm, @disassembler, are you sure you tested the latest commit?
```
nix-shell:~/cardano-node]$ git diff

[nix-shell:~/cardano-node]$ grev
7d998a6155cf16eecb698d8b6fbaefbe34e66591 Tue Sep 10 21:06:48 2019 +0300 cardano-cluster-service:  a flock of cardano-node@NODE-ID services

[nix-shell:~/cardano-node]$ nix-build -A scripts.mainnet.node
/nix/store/c586rsqmd211gjlxbciacax7x4k317dr-cardano-node-mainnet

[nix-shell:~/cardano-node]$ /nix/store/c586rsqmd211gjlxbciacax7x4k317dr-cardano-node-mainnet
Starting cardano-node with --node-id 0 --port 3001 --topology /nix/store/g8fb0sjaq39dbxnvv35dqcwafgvc26vx-topology.yaml --log-config /nix/store/jw29jyvvzdfdwccdc96qckwjbywva6c5-log-configuration.yaml
[iohk.cardano.node.CoreId 0:Debug:4] [2019-09-11 20:17:43.67 UTC] tracing verbosity = normal
[iohk.cardano.node.CoreId 0:Debug:4] [2019-09-11 20:17:44.14 UTC] System started at SystemStart {getSystemStart = 2017-09-23 21:44:51 UTC}
```

**Pull request merged**